src/ints/mouse_dos_driver.cpp

```diff
@@ -0,0 +1,1440 @@
+            IO_Write(0x3CF, gfxReg3CE[i]);
+        }
+
+        IO_Write(0x3C4, 2);
+        IO_Write(0x3C5, gfxReg3C5);
+        IO_Write(0x3C4, index3C4);
+    }
+}
+
+static void ClipCursorArea(int16_t &x1, int16_t &x2, int16_t &y1, int16_t &y2,
+                           uint16_t &addx1, uint16_t &addx2, uint16_t &addy)
+{
+    addx1 = 0;
+    addx2 = 0;
+    addy  = 0;
+    // Clip up
+    if (y1 < 0) {
+        addy = static_cast<uint16_t>(addy - y1);
+        y1 = 0;
+    }
+    // Clip down
+    if (y2 > state.clipy) {
+        y2 = state.clipy;
+    };
+    // Clip left
+    if (x1 < 0) {
+        addx1 = static_cast<uint16_t>(addx1 - x1);
+        x1 = 0;
+    };
+    // Clip right
+    if (x2 > state.clipx) {
+        addx2 = static_cast<uint16_t>(x2 - state.clipx);
+        x2    = state.clipx;
+    };
+}
+
+static void RestoreCursorBackground()
+{
+    if (state.hidden || state.inhibit_draw || !state.background)
+        return;
+
+    SaveVgaRegisters();
+
+    // Restore background
+    int16_t x, y;
+    uint16_t addx1, addx2, addy;
+    uint16_t dataPos = 0;
+    int16_t x1       = static_cast<int16_t>(state.backposx);
+    int16_t y1       = static_cast<int16_t>(state.backposy);
+    int16_t x2       = static_cast<int16_t>(x1 + CURSOR_SIZE_X - 1);
+    int16_t y2       = static_cast<int16_t>(y1 + CURSOR_SIZE_Y - 1);
+
+    ClipCursorArea(x1, x2, y1, y2, addx1, addx2, addy);
+
+    dataPos = static_cast<uint16_t>(addy * CURSOR_SIZE_X);
+    for (y = y1; y <= y2; y++) {
+        dataPos = static_cast<uint16_t>(dataPos + addx1);
+        for (x = x1; x <= x2; x++) {
+            INT10_PutPixel(static_cast<uint16_t>(x),
+                           static_cast<uint16_t>(y),
+                           state.page,
+                           state.backData[dataPos++]);
+        };
+        dataPos = static_cast<uint16_t>(dataPos + addx2);
+    };
+    state.background = false;
+
+    RestoreVgaRegisters();
+}
+
+void MOUSEDOS_DrawCursor()
+{
+    if (state.hidden || state.inhibit_draw)
+        return;
+    INT10_SetCurMode();
+    // In Textmode ?
+    if (CurMode->type == M_TEXT) {
+        DrawCursorText();
+        return;
+    }
+
+    // Check video page. Seems to be ignored for text mode.
+    // hence the text mode handled above this
+    // >>> removed because BIOS page is not actual page in some cases, e.g.
+    // QQP games
+    //    if (real_readb(BIOSMEM_SEG,BIOSMEM_CURRENT_PAGE) != state.page)
+    //    return;
+
+    // Check if cursor in update region
+    /*    if ((GetPosX() >= state.update_region_x[0]) && (GetPosY() <=
+       state.update_region_x[1]) && (GetPosY() >= state.update_region_y[0])
+       && (GETPOS_Y <= state.update_region_y[1])) { if
+       (CurMode->type==M_TEXT16) RestoreCursorBackgroundText(); else
+                RestoreCursorBackground();
+            mouse.shown--;
+            return;
+        }
+       */ /*Not sure yet what to do update region should be set to ??? */
```

Comment:
Are these some notes for yourself? Will they go away in the next PR or something?

Reply:
This is just moved from `mouse.cpp`. I haven't investigated it yet. For now I just want to have DOS part and PS/2 part in separate files, so that when improving DOS part I don;'t have to worry that I break something PS/2 related.